### PR TITLE
Created a Scientific Calculator in a file named scal.py

### PR DESCRIPTION
**Pull request by NikhilJ2005**
#81 Issue Number 

📋 Scientific Calculator Feature Enhancement
📝 Summary
This pull request introduces a scientific calculator to the Streamlit app, implementing various mathematical functions, trigonometric calculations, and angle mode toggling. This addition provides users with an intuitive interface for performing scientific computations directly within the app.

🚀 Features Added
Basic Calculator Operations: Addition, subtraction, multiplication, and division using intuitive symbols (e.g., ×, ÷).
Scientific Functions:
Trigonometric functions (sin, cos, tan), which adapt to the current angle mode (Degree/Radian).
Exponentiation (^), logarithmic functions (log, ln), square root (√), and the mathematical constants (π, e).
Angle Mode Toggle: Switch between Degree and Radian modes for trigonometric calculations.
Expression Display: Dynamically displays the current input expression and result in the display box.
Utility Buttons:
AC: Clears the entire expression.
DC: Deletes the last character in the expression.
Mode: Toggles between Degree and Radian modes.
Equals (=): Evaluates the current expression using a safe evaluation method.
Automatic Multiplication Insertion: Automatically inserts a multiplication symbol when needed, such as before parentheses or mathematical constants (π, e).
🛠️ Implementation Details
Session State Management: Utilizes Streamlit's session state to keep track of the current expression, display content, and angle mode.
Mathematical Expression Parsing: Implements a safe_dict with custom trigonometric functions to safely evaluate the input using Python's eval function.
Button Layout: Organizes calculator buttons using Streamlit's columns for a grid-like appearance, with specific functionality mapped to each button.
🎨 Adjustments
Adjusted symbols to prevent potential Markdown parsing issues.
Improved user interface to intuitively represent scientific calculator functions.
🔍 Testing
Validated button interactions for basic arithmetic operations.
Tested trigonometric calculations in both Degree and Radian modes.
Ensured that the display updates correctly upon expression evaluation, clearing, and character deletion.

WhatsApp Image 2024-10-04 at 7 11 26 PM
WhatsApp Image 2024-10-04 at 7 11 26 PM (1)
WhatsApp Image 2024-10-04 at 7 11 20 PM
WhatsApp Image 2024-10-04 at 7 11 13 PM

### CODE REVIEW

**Comment by jaidh01** [comment]
Hey @NikhilJ2005, does your project run properly with the streamlit application?

**Review by Avdhesh-Varshney — CHANGES_REQUESTED** [review]
Follow the guidelines and attach a full screen working video.

**Comment by NikhilJ2005** [comment]
https://youtu.be/nSQAcrjB2qk?si=_YYjtDTMSN2w9Dnz

Is this enough?.

**Comment by Avdhesh-Varshney** [comment]
I want to show your calculator whether it is properly integrated with the jarvis or not. So, prepare a video from login upto 2-3 test cases on calculator. That's enough

**Comment by NikhilJ2005** [comment]
https://youtu.be/nCZ0ZDUnjt0?si=_FAkCu05RBNBbpla. Done

File: src/utils/functions.py
Comment (Avdhesh-Varshney) [inline src/utils/functions.py:1]: Undo these changes.

File: scal.py
Comment (Avdhesh-Varshney) [inline scal.py:1]: Keep this file in [`StudyPrograms`](https://github.com/Avdhesh-Varshney/Jarvis/tree/main/src/apps/pages/programs/StudyPrograms) folder by renaming to `scientificCalculator`

**Review by Avdhesh-Varshney — CHANGES_REQUESTED** [review]
Follow the guidelines

**Comment by NikhilJ2005** [comment]
okay

**Comment by Avdhesh-Varshney** [comment]
What's the update?

**Comment by NikhilJ2005** [comment]
Can you assign some one else for this issue , since semester exam has started too much busy.

**Pull request closed without merging**